test/boost/sstable_3_x_test.cc

```diff
@@ -5259,3 +5259,48 @@ SEASTAR_THREAD_TEST_CASE(test_sstable_log_too_many_rows) {
     test_sstable_log_too_many_rows_f(random, (random + 1), false);
     test_sstable_log_too_many_rows_f((random + 1), random, true);
 }
+
+// The following test runs on test/resource/sstables/3.x/uncompressed/legacy_udt_in_collection
+// It was created using Scylla 3.0.x using the following CQL statements:
+//
+// CREATE KEYSPACE ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
+// CREATE TYPE ks.ut (a int, b int);
+// CREATE TABLE ks.t (pk int PRIMARY KEY, m map<int, frozen<ut>>) WITH compression = {};
```

Comment:
Could you please also add a test that checks frozen<map<int, frozen<ut>>>?

Reply:
It would be better if the test has covered all the collection types.

Reply:
Also nested collection please.

Reply:
Testing other collection types is redundant, this test + a quick glance at `freeze_types_in_collections` should be enough evidence that this thing works IMHO.

But I agree that a nested frozen collection test could be good.

Reply:
I meant both nested collection and a top-level frozen collection since you said it's affected as well. I don't think it's redundant to test all the types of collections. If you made a mistake in the set then this test wouldn't catch that. Same situation with someone changing the code later. The person would run a test and thought all is good even when the mistake was introduced into the handling of a list.

Reply:
I mean, there is really no difference in `freeze_types_in_collection` between list/set/map handling.
If someone broke sets but not maps, they would have to modify just the set part of this function. Which means they would immediately notice that something is wrong. (or they would have to be really careless)

we can make a short proof:
1. all other code is independent of the particular collection affected
2. test checks that maps are handled correctly
3. `freeze_types_in_collection` handles sets and lists the same way it handles maps

=> sets and lists are handled correctly too

(and I'm saying that without actually checking sets and lists manually. But I'll do this and I'll be really ashamed if they won't work)

Reply:
But I can add the test, no problem ;)

Reply:
The tests are also there to help when a programmer fails to notice - even the obvious stuff ;).

Reply:
Thanks!